engine/runtime-compiler/src/main/scala/org/enso/compiler/pass/analyse/AliasAnalysis.scala

```diff
@@ -796,14 +798,37 @@ case object AliasAnalysis extends IRPass {
     )
   }
 
+  /** Performs alias analysis on a if then else expression.
+    *
+    * @param ir          the expression to analyse
+    * @param graph       the graph in which the analysis is taking place
+    * @param parentScope the scope in which the expression occurs
+    * @return `ir`, possibly with alias analysis information attached
+    */
+  private def analyseIfThenElse(
+    ir: IfThenElse,
+    graph: Graph,
+    parentScope: Scope
+  ): IfThenElse = {
+    val condScope  = parentScope.addChild(flattenToParent = true)
+    val trueScope  = parentScope.addChild(flattenToParent = true)
+    val falseScope = parentScope.addChild(flattenToParent = true)
```

Comment:
These `addChild(flattenToParent = true)` is causing various failures with more complicated structures of `if`s in a single method. For example [this failure is caused by overridden local variables in previous method](https://github.com/enso-org/enso/actions/runs/11530291708/job/32099687962?pr=11365#step:7:2132).
- let's record the status as [addChildFlattenToParent](https://github.com/JaroslavTulach/enso/tree/addChildFlattenToParent)
- Let's try without flattening to parent in this PR